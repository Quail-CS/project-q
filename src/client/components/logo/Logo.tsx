/**
 * @module Logo.tsx
 * @description Branding Logo Component
 */

const brandingLogo = require('../../assets/brandingLogo.png')

import * as React from 'react';

interface LogoProps {
  className: string,
}

const Logo = (props: LogoProps) => {
  const caption = "Welcome to Ethiq";
  const { className } = props;
  return (
    <img alt={caption} className={className} src={brandingLogo} />
  )
};

export default Logo;

