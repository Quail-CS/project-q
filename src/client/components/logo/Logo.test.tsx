import * as React from 'react';
import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import * as Renderer from 'react-test-renderer';

import Logo from './Logo';

const brandingLogo = require('../assets/brandingLogo.png')

Enzyme.configure({ adapter: new Adapter() });

describe('Loading Component: ', () => {
  const stubClassName = 'logo-image';

  it('matches the snapshot', () => {
    const tree = Renderer.create(<Logo className={stubClassName} />).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('renders an <img> with class onboarding', () => {
    const onboardingWrapper: Enzyme.ShallowWrapper = Enzyme.shallow(<Logo className={stubClassName} />);
    expect(onboardingWrapper.hasClass(stubClassName)).toEqual(true);
    expect(onboardingWrapper.type()).toEqual('img');
  })

  it('renders an <img> with alt "Welcome to Ethiq" and src "brandinglogo.png"', () => {
    const onboardingWrapper: Enzyme.ShallowWrapper = Enzyme.shallow(<Logo className={stubClassName} />);
    expect(onboardingWrapper.prop('alt')).toEqual('Welcome to Ethiq');
    expect(onboardingWrapper.prop('src')).toEqual(brandingLogo);
  })
});
