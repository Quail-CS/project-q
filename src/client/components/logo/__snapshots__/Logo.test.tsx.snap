// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component:  matches the snapshot 1`] = `
<img
  alt="Welcome to Ethiq"
  className="logo-image"
  src={
    Object {
      "process": [Function],
    }
  }
/>
`;
